File: frontend/src/api/actions.ts
import { apiClient } from './client'
import type { Action } from '@/types'

export const actionsApi = {
  create: (data: { scan_id: string; action_type: string; agency?: string; priority?: string; note?: string }) =>
    apiClient.post<Action>('/actions', data).then((r) => r.data),

  list: (params?: { scan_id?: string; page?: number }) =>
    apiClient.get<Action[]>('/actions', { params }).then((r) => r.data),

  updateStatus: (id: string, status: string) =>
    apiClient.patch<Action>(`/actions/${id}`, { status }).then((r) => r.data),
}

File: backend/app/routers/actions.py
import uuid
from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy.ext.asyncio import AsyncSession
from app.database import get_db
from app.dependencies import get_current_user
from app.models.user import User
from app.schemas.action import ActionCreate, ActionResponse, ActionStatusUpdate
from app.services import action_service

router = APIRouter(prefix="/api/v1/actions", tags=["actions"])


@router.post("", response_model=ActionResponse, status_code=status.HTTP_201_CREATED)
async def create_action(
    data: ActionCreate,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return await action_service.create_action(db, data, current_user.id)


@router.get("", response_model=list[ActionResponse])
async def list_actions(
    scan_id: uuid.UUID | None = None,
    page: int = Query(1, ge=1),
    db: AsyncSession = Depends(get_db),
    _: User = Depends(get_current_user),
):
    return await action_service.list_actions(db, scan_id, page)


@router.patch("/{action_id}", response_model=ActionResponse)
async def update_action(
    action_id: uuid.UUID,
    data: ActionStatusUpdate,
    db: AsyncSession = Depends(get_db),
    _: User = Depends(get_current_user),
):
    action = await action_service.update_action_status(db, action_id, data.status)
    if not action:
        raise HTTPException(status_code=404, detail="Amal topilmadi")
    return action
